text/0499-appconfig-constructs.md

```diff
@@ -0,0 +1,422 @@
+  * Amazon CloudWatch Evidently A/B testing: This extension allows your application to assign variations to user sessions locally instead of by calling the EvaluateFeature operation.
+  * AWS AppConfig deployment events to EventBridge: This extension sends events to the EventBridge default event bus when a configuration is deployed.
+  * AWS AppConfig deployment events to Amazon Simple Notification Service (Amazon SNS): This extension sends messages to an Amazon SNS topic that you specify when a configuration is deployed.
+  * AWS AppConfig deployment events to Amazon Simple Queue Service (Amazon SQS): This extension enqueues messages into your Amazon SQS queue when a configuration is deployed.
+  * Integration extension—Atlassian Jira: This extensions allows AWS AppConfig to create and update issues whenever you make changes to a feature flag.
+
+You can associate these types of tasks with AWS AppConfig applications, environments, and configuration profiles.
+
+For more information about extensions, see [About AWS AppConfig extensions](https://docs.aws.amazon.com/appconfig/latest/userguide/working-with-appconfig-extensions-about.html).
+
+### Example
+```ts
+const extension = new Extension(this, 'MyExtension', {
+  actions: [
+    new Action(this, 'MyAction', {
+      actionPointEvent: <IFunction|IQueue|IBus|ITopic>,  // ex. ActionPointEvent.fromFunction(<IFunction>)
+      actionPoint: ActionPoint.ON_DEPLOYMENT_COMPLETE,
+    }),
+  ],
+
+  // optional
+  name: 'ExtensionName',
+  resource: <IApplciation|IConfigurationProfile|IEnvironment>,   // ex. ExtensionResource.fromApplication(<IApplication>)
+});
+
+// can also add more extension associations
+extension.addExtensionAssociation({
+  resource: <IApplciation|IConfigurationProfile|IEnvironment>,
+});
+```
+
+To customize existing AWS AppConfig authored extensions, we can do the following as an example.
+
+```ts
+const extension = Extension.customAppConfigDeploymentEventsToAmazonSQS({
+  actionPoints: [
+    ActionPoint.ON_DEPLOYMENT_START,
+    ActionPoint.ON_DEPLOYMENT_BAKING,
+    ActionPoint.ON_DEPLOYMENT_COMPLETE,
+  ],
+});
+```
+
+## Higher level resources
+
+This section details how we will implement our constructs that combine multiple different resources into one.
+
+# Configuration
+
+A configuration is a higher level construct that can either be a HostedConfiguration (stored internally through AppConfig) or a SourcedConfiguration (stored in S3, Secrets Manager, SSM Parameter, SSM Document, or Code Pipeline). This construct will be used as a building block to deploy configuration through the AppConfig construct (detailed in the next section). For a HostedConfiguration, this construct will combine both a configuration profile and a hosted configuration version.
```

Comment:
The thought for the lower-level `ConfigurationProfile` and `HostedConfigurationVersion` is so that users can work with the low-level stuff if they want to (for example to import existing resources). If they have different use cases than using the high level constructs, they will have the flexibility to work directly with the base resources